lib/tether/ops_linux.go

```diff
@@ -326,6 +328,37 @@ func renameLink(t Netlink, link netlink.Link, slot int32, endpoint *NetworkEndpo
 	return link, nil
 }
 
+func gratuitous(t Netlink, link netlink.Link, endpoint *NetworkEndpoint) error {
+	if ip.IsUnspecifiedIP(endpoint.Assigned.IP) {
+		// refusing to broadcast unspecified IP
+		return nil
+	}
+
+	intf, err := net.InterfaceByIndex(link.Attrs().Index)
```

Comment:
Is link.Attrs().Index actually the same as slot in here? `nl.LinkBySlot(int32(slot))` Just curious.

Reply:
No. `LinkBySlot` is getting a link based on it's PCI slot number, whereas `Index` is the internal Linux indexing for the interfaces.